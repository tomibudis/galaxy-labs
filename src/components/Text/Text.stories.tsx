import React from 'react';

import { Meta, ComponentStory } from '@storybook/react';

import Text from './Text';

export default {
  title: 'Components/Text',
  component: Text,
  argsTypes: {
    children: {
      table: {
        disable: true,
      },
    },
  },
} as Meta<typeof Text>;

const Template: ComponentStory<typeof Text> = (args): JSX.Element => {
  return <Text {...args}>{args.children}</Text>;
};

export const BasicText = Template.bind({});
BasicText.args = {
  children:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur suscipit purus nec sem pulvinar euismod. Proin sit amet lacus cursus, sagittis ante non, ultrices diam. Nunc vehicula lectus eu leo vestibulum aliquet. Vivamus sagittis sapien lorem, quis fringilla felis gravida a. Donec id nulla massa. Morbi et ante ullamcorper nulla tristique feugiat in eget lectus. Suspendisse in turpis eget massa ultricies malesuada vel quis ipsum. Aenean auctor, nibh tincidunt dapibus laoreet, eros velit iaculis diam, ut varius risus mauris eget ipsum',
};

export const ColorText = Template.bind({});
ColorText.args = {
  children: 'Hello here is a text component',
  className: 'text-danger',
};
