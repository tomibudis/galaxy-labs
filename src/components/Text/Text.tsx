import React from 'react';

import PropTypes from 'prop-types';
import cx from 'classnames';

import { TextProps } from '#interface/index';

const Text: React.FC<TextProps> = ({ className, children }): JSX.Element => {
  return <p className={cx('mb-0', className)}>{children}</p>;
};

Text.propTypes = {
  /** Additional classname to define color of text */
  className: PropTypes.string,
  /** Children ReactNode */
  children: PropTypes.node.isRequired,
};

export default Text;
