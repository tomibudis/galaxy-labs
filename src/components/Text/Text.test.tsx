import React from 'react';
import { cleanup, render } from '@testing-library/react';

import Text from './Text';

afterEach(() => {
  cleanup();
});

const TextComponent = (props): JSX.Element => {
  return <Text {...props}>{props?.children || 'Text Component'}</Text>;
};

describe('<Text />', () => {
  it('render component without crashing', () => {
    render(<TextComponent />);
    expect(true).toBeTruthy();
  });
  it('match with snapshot file', () => {
    const { baseElement } = render(
      <TextComponent className="text-danger">here a text</TextComponent>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
