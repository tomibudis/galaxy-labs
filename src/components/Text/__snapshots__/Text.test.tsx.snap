// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> match with snapshot file 1`] = `
<body>
  <div>
    <p
      class="mb-0 text-danger"
    >
      here a text
    </p>
  </div>
</body>
`;
